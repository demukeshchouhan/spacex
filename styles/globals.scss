html,
body {
  padding: 0;
  margin: 0;
  font-family: -apple-system, BlinkMacSystemFont, Segoe UI, Roboto, Oxygen,
    Ubuntu, Cantarell, Fira Sans, Droid Sans, Helvetica Neue, sans-serif;
  background-color: #e9e9e9;
  overflow: hidden;
}

a {
  color: inherit;
  text-decoration: none;
}

* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  list-style: none;
  text-decoration: none;
  font-family: "Josefin Sans", sans-serif;
}

.row {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
}

@media only screen and (max-width: 600px) {
  html,
  body {
    overflow: scroll;
  }
  img {
    height: 100% !important;
    max-width: 100% !important;
  }
}
