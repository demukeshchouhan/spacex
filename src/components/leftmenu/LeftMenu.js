import React, { Component } from "react";
import get from "lodash/get";
import { connect } from "react-redux";

import { fetchLaunches } from "../../redux/actions/launches";
import Button from "../common/Button";
import styles from "./leftmenu.module.scss";

class LeftMenu extends Component {
  state = {
    isLanding: true,
    launchYear: get(this.props, "data[0].launch_year"),
    launchesData: get(this.props, "data"),
  };

  handleLaunchClick = (evt) => {
    this.handleCommonClick(evt, {
      prevState: "isLaunch",
      nextState: "launchValue",
    });
  };

  handleLandingClick = (evt) => {
    this.handleCommonClick(evt, {
      prevState: "isLanding",
      nextState: "landingValue",
    });
  };

  handleLaunchYearCLick = (evt) => {
    this.handleCommonClick(evt, {
      prevState: "isYear",
      nextState: "launchYear",
    });
  };

  handleCommonClick = (evt, params) => {
    const { value } = evt.target;
    const { prevState, nextState } = params;
    this.setState({ [prevState]: value, [nextState]: value });
    this.props.fetchLaunches({
      ...this.state,
      [prevState]: true,
      [nextState]: value,
    });
  };

  render() {
    const { data } = this.props;
    return (
      <div className={styles.leftmenu}>
        <div className={styles.linkPanel}>
          <h2>Filters</h2>
          <h3 className={styles.subHeading}>
            <span>Launch Year</span>
          </h3>
          <ul className={styles.yearList}>
            {data &&
              data.map((item, index) => {
                if ((index + 1) % 3 == 0) {
                  return <br key={item.flight_number} />;
                }
                return (
                  <li key={item.flight_number}>
                    <Button
                      name={item.launch_year}
                      active={this.state.launchYear === item.launch_year}
                      value={item.launch_year}
                      onClick={this.handleLaunchYearCLick}
                    />
                  </li>
                );
              })}
          </ul>
          <h3 className={styles.subHeading}>
            <span>SuccessFul Launch</span>
          </h3>
          <ul className={styles.yearList}>
            <li>
              <Button
                onClick={this.handleLaunchClick}
                name="True"
                value={"true"}
                active={this.state.isLaunch === "true"}
              />
            </li>
            <li>
              <Button
                onClick={this.handleLaunchClick}
                name="False"
                value={"false"}
                active={this.state.isLaunch === "false"}
              />
            </li>
          </ul>
          <h3 className={styles.subHeading}>
            <span>SuccessFul Landing</span>
          </h3>
          <ul className={styles.yearList}>
            <li>
              <Button
                onClick={this.handleLandingClick}
                name="True"
                value={"true"}
                active={this.state.isLanding === "true"}
              />
            </li>
            <li>
              <Button
                onClick={this.handleLandingClick}
                name="False"
                value={"false"}
                active={this.state.isLanding === "false"}
              />
            </li>
          </ul>
        </div>
      </div>
    );
  }
}

const mapState = () => {
  return {};
};

const mapProps = {
  fetchLaunches,
};

export default connect(mapState, mapProps)(LeftMenu);
