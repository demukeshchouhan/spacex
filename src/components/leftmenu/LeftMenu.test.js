import React from "react";
import { shallow, mount } from "enzyme";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";

import LeftMenu from "./LeftMenu";
import styles from "./leftmenu.module.scss";

let wrapper = null;
let store = configureMockStore();
let defaultProps = {
  className: styles.leftmenu,
  data: [],
};
const mockStore = store({});
let LeftMenuWithProvider = () => (
  <Provider store={mockStore}>
    <LeftMenu {...defaultProps} />
  </Provider>
);

beforeEach(() => {
  wrapper = shallow(LeftMenuWithProvider()).dive();
});

describe("LeftMenu", () => {
  it("should render", () => {
    expect(wrapper).toBeDefined();
  });

  it("should match snapshot", () => {
    const wrapper = mount(LeftMenuWithProvider());
    expect(wrapper).toMatchSnapshot();
  });

  it(`should have className ${styles.leftmenu}`, () => {
    expect(wrapper.props().className).toBe(styles.leftmenu);
  });

  it("should have data props", () => {
    expect(wrapper.props().data).toBeTruthy();
  });

  it("should have handleCommonClick method defined", () => {
    const wrapper = mount(LeftMenuWithProvider());
    expect(wrapper.find("LeftMenu").instance().handleCommonClick).toBeDefined();
  });
  it("should have handleLaunchClick method defined", () => {
    const wrapper = mount(LeftMenuWithProvider());
    expect(wrapper.find("LeftMenu").instance().handleLaunchClick).toBeDefined();
  });
  it("should have handleLandingClick method defined", () => {
    const wrapper = mount(LeftMenuWithProvider());
    expect(
      wrapper.find("LeftMenu").instance().handleLandingClick
    ).toBeDefined();
  });
  it("should have handleLaunchYearCLick method defined", () => {
    const wrapper = mount(LeftMenuWithProvider());
    expect(
      wrapper.find("LeftMenu").instance().handleLaunchYearCLick
    ).toBeDefined();
  });
});
