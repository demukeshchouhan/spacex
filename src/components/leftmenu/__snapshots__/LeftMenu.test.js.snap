// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeftMenu should match snapshot 1`] = `ReactWrapper {}`;
